src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -998,7 +999,9 @@ struct FixedKVLeafNode
   // 2. The child extent is stable
   bool is_child_stable(uint16_t pos) const final {
     auto child = this->children[pos];
-    if (is_valid_child_ptr(child)) {
+    if (is_reserved_ptr(child)) {
+      return true;
+    } else if (is_valid_child_ptr(child)) {
       ceph_assert(child->is_logical());
       return child->is_stable();
```

Comment:
If I'm not missing anything -- `is_child_stable(pos)` will return true even if the child in the according transaction is mutation-pending, which is incorrect, right?

Same to the below case.

Reply:
> If I'm not missing anything -- `is_child_stable(pos)` will return true even if the child in the according transaction is mutation-pending, which is incorrect, right?

Do you recall why we consider mutation-pending extents stable? From https://github.com/ceph/ceph/pull/54351, it seems we deliberately distinguished `stable` and `stable_written`.

> 
> Same to the below case.

Reply:
> If I'm not missing anything -- `is_child_stable(pos)` will return true even if the child in the according transaction is mutation-pending, which is incorrect, right?
> 
> Same to the below case.

I think `stable child` means that the child is not directly linked to a pending lba parent, maybe the naming is not appropriate.

Reply:
The issue seems orthogonal.

Regarding to `is_stable()` vs `is_stable_written()`. If I understand correctly, the mutation-pending extent is only considerred stable if it is marked to have pending I/O upon `prepare_record()`.

I recall it is because of the caveats that there are some checks happening in the middle of `prepare_record()`, while not all the links to the mutation-pending extents are corrected yet. (worth adding some comments?)

Reply:
@xxhdx1985126 I think `FixedKVLeafNode::is_child_stable(uint16_t pos)` may need to be queried before the transaction is committed. According to the above discussions, this means if the extent in transaction referred by the pin is mutation-pending, `pin->is_stable()` will return true, which should be wrong.

Or am I missing something?

Reply:
> @xxhdx1985126 I think `FixedKVLeafNode::is_child_stable(uint16_t pos)` may need to be queried before the transaction is committed. According to the above discussions, this means if the extent in transaction referred by the pin is mutation-pending, `pin->is_stable()` will return true, which should be wrong.

I think this is possible, for example:

1. `trans.1` mutated extent `E` and got submitted;
2. Before `trans.1` is committed, other transactions that retrieve `E`'s lba mapping and tries to determine the mapping's stability will see `E` as mutation-pending and under-io, which I think should be considered correct.

What do you think?
> 
> Or am I missing something?